### PR TITLE
missing bits from last pr merge and some

### PR DESCRIPTION
**Pull request by nitincchauhan**

### CODE REVIEW

File: src/main/java/com/zaxxer/hikari/pool/PoolBase.java [hunk 1]
```diff
@@ -315,7 +315,8 @@ Connection newConnection() throws Exception
       }
       catch (Exception e) {
          lastConnectionFailure.set(e);
-         quietlyCloseConnection(connection, "(Failed to create/set connection)");
+         String reason = e.getMessage();
+         quietlyCloseConnection(connection, reason != null ? reason : "(Failed to create/set connection)");
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/PoolBase.java]: Intentionally not merged.

File: src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java [hunk 2]
```diff
@@ -248,8 +248,10 @@ public final void close() throws SQLException
             }
          }
          finally {
-            delegate = ClosedConnection.CLOSED_CONNECTION;
-            poolEntry.recycle(lastAccess);
+            if (delegate != ClosedConnection.CLOSED_CONNECTION) {
+               delegate = ClosedConnection.CLOSED_CONNECTION;
+               poolEntry.recycle(lastAccess);
+            }
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java]: Intentionally not merged.

File: src/main/java/com/zaxxer/hikari/util/PropertyElf.java [hunk 3]
```diff
@@ -41,7 +41,7 @@
 {
    private static final Logger LOGGER = LoggerFactory.getLogger(PropertyElf.class);
 
-   private static final Pattern GETTER_PATTERN = Pattern.compile("(get|is)[A-Z].+");
+   private static final Matcher METHOD_MATCHER = Pattern.compile("(get|is)[A-Z].+").matcher("");
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/util/PropertyElf.java]: Matcher is not thread-safe and cannot be held as a static member.

File: src/main/java/com/zaxxer/hikari/pool/PoolBase.java (hunk 1 above)
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/PoolBase.java]: it is better to know 'actual reason', this text is more important as there can be many reason and is used for logging purpose to help debug/trace problem

File: src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java (hunk 2 above)
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java]: this avoids 'possibility' of adding to the bag and worse 'giving user' a ClosedConnection.CLOSED_CONNECTION.

File: src/main/java/com/zaxxer/hikari/util/PropertyElf.java (hunk 3 above)
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/util/PropertyElf.java]: good catch

**Pull request closed without merging**